geocodingsearchv7/batchgeocoder_test.go

```diff
@@ -0,0 +1,108 @@
+type BatchGeocodingMock struct {
+	responseStatus int
+	responseBody   geocodingsearchv7.BatchGeocoderResponse
+}
+
+func (c *BatchGeocodingMock) Do(req *http.Request) (*http.Response, error) {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json")
+	b, err := json.Marshal(c.responseBody)
+	if err != nil {
+		return nil, err
+	}
+	r := bytes.NewReader(b)
+	return &http.Response{
+		StatusCode:    c.responseStatus,
+		Header:        headers,
+		Body:          io.NopCloser(r),
+		ContentLength: int64(len(b)),
+	}, nil
+}
+
+func TestBatchGeocodingService_BatchGeocode(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	t.Run(
+		"when both addresses and queries is nil, then return InvalidArgument ",
+		func(t *testing.T) {
+			t.Parallel()
+			httpClient := BatchGeocodingMock{responseBody: geocodingsearchv7.BatchGeocoderResponse{}, responseStatus: 200}
+			routingClient := geocodingsearchv7.NewClient(&httpClient)
+
+			_, err := routingClient.BatchGeocoder.BatchGeocoderUpload(ctx, &geocodingsearchv7.BatchGeocoderUploadRequest{
+				Addresses: nil, // Addresses is nil
+				Queries:   nil, // Queries is nil
+			})
+			assert.ErrorContains(t, err, "InvalidArgument")
+		},
+	)
+	t.Run(
+		"when both addresses and queries is not, then return InvalidArgument ",
+		func(t *testing.T) {
+			t.Parallel()
+			httpClient := BatchGeocodingMock{responseBody: geocodingsearchv7.BatchGeocoderResponse{}, responseStatus: 200}
+			routingClient := geocodingsearchv7.NewClient(&httpClient)
+
+			_, err := routingClient.BatchGeocoder.BatchGeocoderUpload(ctx, &geocodingsearchv7.BatchGeocoderUploadRequest{
+				Addresses: []*geocodingsearchv7.AddressRequest{},
+				Queries:   []*geocodingsearchv7.QueryString{},
+			})
+			assert.ErrorContains(t, err, "InvalidArgument")
+		},
+	)
+}
+
+func TestBatchGeocodingService_BatchGeocodeStatus(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	t.Run(
+		"when requestID is empty, then return InvalidArgument",
+		func(t *testing.T) {
+			t.Parallel()
+			httpClient := BatchGeocodingMock{responseBody: geocodingsearchv7.BatchGeocoderResponse{}, responseStatus: 200}
+			routingClient := geocodingsearchv7.NewClient(&httpClient)
+
+			_, err := routingClient.BatchGeocoder.BatchGeocoderStatus(ctx, &geocodingsearchv7.BatchGeocoderStatusRequest{
+				RequestID: "",
+			})
+			assert.ErrorContains(t, err, "InvalidArgument")
+		},
+	)
+}
+
+func TestBatchGeocodingService_BatchGeocodeDownload(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	t.Run(
+		"when requestID is empty, then return InvalidArgument",
+		func(t *testing.T) {
+			t.Parallel()
+			httpClient := BatchGeocodingMock{responseBody: geocodingsearchv7.BatchGeocoderResponse{}, responseStatus: 200}
+			routingClient := geocodingsearchv7.NewClient(&httpClient)
+
+			err := routingClient.BatchGeocoder.BatchGeocoderDownload(
+				ctx,
+				&geocodingsearchv7.BatchGeocoderDownloadRequest{RequestID: ""}, // RequestID is empty
```

Comment:
```suggestion
				&geocodingsearchv7.BatchGeocoderDownloadRequest{RequestID: ""},
```
Superfluous comment?